avm/ptn/app/iaas-vm-cosmosdb-tier4/main.bicep

```diff
@@ -0,0 +1,845 @@
+metadata name = 'IaaS VM with CosmosDB Tier 4'
+metadata description = 'Creates an IaaS VM with CosmosDB Tier 4 resiliency configuration.'
+
+@export()
+@description('The type of a subnet.')
+type subnetType = {
+  @description('Required. The name of the subnet.')
+  name: string
+
+  @description('Required. The address prefix for the subnet.')
+  addressPrefix: string
+}
```

Comment:
As usual, not a must, but usually most contributors place the types at the very bottom of the file like

```bicep

// =============== //
//   Definitions   //
// =============== //

@export()
@description('The type of a subnet.')
type subnetType = {
  @description('Required. The name of the subnet.')
  name: string

  @description('Required. The address prefix for the subnet.')
  addressPrefix: string
}
```
Kinda like an appendix. Feel free to resolve if you'd prefer it up here in the parameters. 💪